src/odata-to-abstract-sql.ts

```diff
@@ -717,8 +780,15 @@ export class OData2AbstractSQL {
 		) {
 			// For update/delete statements we need to use a  style query
 			const subQuery = new Query();
-			subQuery.select.push(referencedIdField);
 			subQuery.fromResource(this, resource);
+			subQuery.addNestedFieldSelect(
+				resource.idField,
+				method === 'PATCH'
+					? '$updateid'
+					: method === 'DELETE'
+						? '$deleteid'
+						: '$upsertid',
```

Comment:
I think it might make more sense to just use eg `$modifyid` to avoid branching on the name, since I don't think there's ever a point where we actually need to differentiate between update/delete/upsert in the naming since they never intermix?

Reply:
That's what I wanted from the first place :) :tada: